tests/integration/test_utils/data_source_config/base.py

```diff
@@ -84,15 +84,27 @@ def __hash__(self) -> int:
 
 
 _ConfigT = TypeVar("_ConfigT", bound=DataSourceTestConfig)
+_AssetT = TypeVar("_AssetT", bound=DataAsset)
 
 
-class BatchTestSetup(ABC, Generic[_ConfigT]):
+class BatchTestSetup(ABC, Generic[_ConfigT, _AssetT]):
     """ABC for classes that set up and tear down batches."""
 
     def __init__(self, config: _ConfigT, data: pd.DataFrame) -> None:
         self.config = config
         self.data = data
 
+    @cached_property
+    def name(self) -> str:
```

Comment:
Before we created one in the `make_batch` methods, but I moved it to a cached property for reuse. It doesn't really matter that we use the same name across datasource/asset/batch def, but I figure why not?